### PR TITLE
[Dynamo] Allow `torch.cond()` to handle emply arguments

### LINKED ISSUE #138150: torch.cond should support omitting arguments to pass in when it is empty
### 🐛 Describe the bug

This program fails:

```
import torch

@torch.compile(fullgraph=True)
def foo(x, y, z):
    def f():
        return y + 2
    def g():
        return z + 1
    return torch.cond(x, f, g)

foo(torch.randn(1), torch.randn(3), torch.randn(3))
```

```
  File "/data/users/ezyang/b/pytorch/torch/_dynamo/symbolic_convert.py", line 948, in inline_user_function_return
    return InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
  File "/data/users/ezyang/b/pytorch/torch/_dynamo/symbolic_convert.py", line 3103, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
  File "/data/users/ezyang/b/pytorch/torch/_dynamo/symbolic_convert.py", line 3157, in inline_call_
    args=[arg.python_type() for arg in args],
  File "/data/users/ezyang/b/pytorch/torch/_dynamo/symbolic_convert.py", line 3157, in <listcomp>
    args=[arg.python_type() for arg in args],
  File "/data/users/ezyang/b/pytorch/torch/_dynamo/variables/base.py", line 213, in python_type
    raise NotImplementedError(f"{self} has no type") from None
torch._dynamo.exc.InternalTorchDynamoError: NotImplementedError: NestedUserFunctionVariable() has no type
```

It is fixed by adding a fourth argument to cond. But this is a pretty terrible graph break message. Also torch.cond should just default fourth argument to empty tuple.

### Versions

main

cc @chauhang @penguinwu @zou3519 @ydwu4 @bdhirsh @yf225

### PR DESCRIPTION
**Pull request by shink**
Fixes #138150

```python
import torch


@torch.compile(fullgraph=True)
def foo(x, y, z):
    def f():
        return y + 2
                                                        
    def g():
        return z + 1
                                                        
    return torch.cond(x, f, g)
                                                        
                                                        
print(foo(torch.zeros(1), torch.ones(1), torch.ones(1))) # tensor([2.])
print(foo(torch.ones(1), torch.ones(1), torch.ones(1))) # tensor([3.])
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

### CODE REVIEW

**Comment by shink** [comment]
@ezyang Please have a look, does this change makes sense to you?

File: torch/_higher_order_ops/cond.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def __call__(self, pred, true_fn, false_fn, operands):
 
 
 @exposed_in("torch")
-def cond(pred, true_fn, false_fn, operands):
+def cond(pred, true_fn, false_fn, operands=()):
```
Comment (ydwu4) [inline torch/_higher_order_ops/cond.py:65]: nit: can help add type annotations like:
```
(pred: Union[bool, torch.Tensor]. true_fn: Callable, false_fn: Callable, operands: Tuple = ())
```

**Review by ydwu4 — COMMENTED** [review]
Sorry for the late. Thanks for working on this!

Some suggestions: can also change the inline documentation as well? Right now, the doc still says operands needs to be provided. Maybe can also add an example to illustrate this behavior.

File: torch/_higher_order_ops/cond.py (hunk 1 above)
Comment (shink) [reply torch/_higher_order_ops/cond.py:65]: Thanks for your reveiw! ah yes I missed the doc! Will finish it now.

**Review by ezyang — APPROVED** [review]

**Review by zou3519 — APPROVED** [review]

**Pull request closed without merging**